src/plugins/intel_cpu/src/nodes/mkldnn_topk_node.cpp

```diff
@@ -99,7 +99,7 @@ struct jit_uni_topk_kernel_f32 : public jit_uni_topk_kernel, public jit_generato
             mov(reg_table, l_table);
 
         data_type = MKLDNNExtensionUtils::IEPrecisionToDataType(jcp_.precision);
-        if (jcp_.layout == TopKLayoutType::topk_blocked && jcp_.topk_innermost)
+        if (!shape_agnostic_alg && jcp_.layout == TopKLayoutType::topk_blocked && jcp_.topk_innermost)
```

Comment:
Though impl before change won't lead to any accuracy or performance issue. blk_stride is not needed for dynamic shape, so it also should not be calculated. It can avoid potential signed integer overflow, when jcp_.sort_stride is uninitialized.

Reply:
Please initialize blk_stride with something so we didn't get uninitialized variable later.

You can do it during declaration `int blk_stride = 0;` on line 212

Reply:
Done. Thanks Andrey!